libraries/AP_Scripting/examples/rev_thr_on_land.lua

```diff
@@ -0,0 +1,181 @@
+local MIN_GROUND_SPEED_              = bind_add_param('GNDSPEED_MIN',            12, 2.5)
+
+-- *****---INTERNAL VARIABLES AND BOOL FLAGS --- DO NOT CHANGE ANYTHING HERE UNLESS MODIFYING THE CODE---*****
+local AP_VERSION_NUM_ = 10 * tonumber(FWVersion:major()) + tonumber(FWVersion:minor())
+local RFND_ROT_DN_ = 25
+local MAX_DIST_ = rangefinder:max_distance_cm_orient(RFND_ROT_DN_) * 0.01
+local MODE_FBWA = 5
+local COUNT_IN_RANGE_ = 0
+local RCMAP_THROTTLE_  = bind_param("RCMAP_THROTTLE")
+local THROT_CHANNEL_ = rc:get_channel(RCMAP_THROTTLE_:get())
+local REV_THRUST_AUX_SW_ = rc:find_channel_for_option(64)
+local REV_THRUST_SW_LOW_PWM_ = 1000
+local REV_THRUST_SW_HIGH_PWM_ = 2000
+local TIME_SEC_ = 0
+local CALLBACK_TIME_ = 200
+local IS_THR_SUP_ = true
+local IS_IN_REV_THRUST_ = false
+local REV_THRUST_SW_ = false
+
+local REV_THRUST_CUR_PWM_ = REV_THRUST_IDLE_PWM_:get() + REV_THRUST_SPOOL_DPWM_:get()
+local TIM_AFTER_TOUCHDN_ = LAND_DET_TIME_:get() * 1000 * 0.01
+
+function UPDATE()
+  if (REV_ENABLE_:get() < 1) then
+    return
+  end
+  if (vehicle:get_mode() == MODE_FBWA and (arming:is_armed())
+      and (rc:get_pwm(RC_ARM_CHANNEL_:get()) > ARM_CHANNEL_HI_PWM_:get())
+      and (THROT_CHANNEL_:norm_input_dz() == 0)) then
+    if (RFND_READING() < THRESH_LAND_HEIGHT_:get()) then
+      if (COUNT_IN_RANGE_ > TIM_AFTER_TOUCHDN_) then
+        TIME_SEC_ = millis():tofloat()
+        return ENGAGE_REV_THRUST()
+      end
+      COUNT_IN_RANGE_ = COUNT_IN_RANGE_ + 1
+    else
+      COUNT_IN_RANGE_ = 0
+    end
+  end
+  return UPDATE, 100 -- check again in 10Hz
+end
+
+function RFND_READING()
+  if rangefinder:has_data_orient(RFND_ROT_DN_) then
+    return rangefinder:distance_cm_orient(RFND_ROT_DN_) * 0.01
+  end
+  return MAX_DIST_
+end
+
+function DISENGAGE_REV_THRUST()
+  THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_:get())
+  REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_LOW_PWM_)
+  gcs:send_text(4, "Reverse Thrust Disengaged")
+  IS_IN_REV_THRUST_ = false
+end
+
+function ENGAGE_REV_THRUST()
+  local groundspeed = ahrs:groundspeed_vector():length()
+
+  if (IS_THR_SUP_ and (vehicle:get_mode() == MODE_FBWA)) then
+    if (millis():tofloat() - TIME_SEC_ < TIME_THROT_WAIT_MS_:get()) then
+      THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_:get()) -- set idle throttle and prepare for reverse, letting the motors stop
+    else
+      REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_HIGH_PWM_)
+      if (REV_THRUST_AUX_SW_:get_aux_switch_pos() == 2) then
+        IS_THR_SUP_ = false
+        REV_THRUST_SW_ = true --set flag for reverse thrust active
+      end
+    end
+  end
+
+  if ((REV_THRUST_AUX_SW_:get_aux_switch_pos() == 2)
+    and (IS_THR_SUP_ == false)
+    and (REV_THRUST_SW_ == true)) then
+    TIME_SEC_ = millis():tofloat()
+    REV_THRUST_SW_ = false
+    IS_IN_REV_THRUST_ = true
+    REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_HIGH_PWM_)
```

Comment:
is this just to re-override so it doesn't time out? Running the aux function directly won't need that.